### PR TITLE
[Abyssea] Fix infusionist history, message and price

### PR DESCRIPTION
**Pull request by Chronos77**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Delete currency when infuse atma

Add message when infuse atma, purge atma, reinfuse previous set and select from history 

Add option 5, used to infuse set of atmas from history

## Steps to test these changes

Infuse atma and see cruor and text

### CODE REVIEW

File: scripts/globals/abyssea/atma.lua [hunk 1]
```diff
@@ -281,12 +283,8 @@ local function updateReinfusedMask(player, atmaSlot, atmaValue)
...
 end
 
-local function updateHistoryMask(player, atmaValue)
-    local currentAtma = getIdByKeyItemId(atmaValue)
-    local histo =
-    {
-        currentAtma
-    }
+local function getHistoryAtmaArray(player)
+    local histo = {}
```
Comment (claywar) [inline scripts/globals/abyssea/atma.lua]: Could this variable be renamed to something a bit more verbose? Something like atmaHistory, etc

File: scripts/globals/abyssea/atma.lua [hunk 2]
```diff
@@ -432,29 +454,56 @@ end
...
     local optionSelected = bit.band(option, 0xF)
     if
-        optionSelected == 1 or
-        optionSelected == 4
-    then -- Infuse Atma or Select from history
+        optionSelected == 1
+    then -- Infuse Atma
         local atma = bit.band(bit.rshift(option, 16), 0xFFFF)
+        local orderKeyItem = getSortedKeysFromArray(xi.atma.atmaMods, false)
+        player:messageSpecial(7446, atmaPrice, orderKeyItem[atma])
```
Comment (claywar) [inline scripts/globals/abyssea/atma.lua]: 7446 needs to be looked up from an enum value

File: scripts/globals/abyssea/atma.lua [hunk 3]
```diff
@@ -432,29 +454,56 @@ end
...
         local slot = bit.band(bit.rshift(option, 16), 0xFF)
         if slot == 4 then -- Purge all atma
-            for atmaSlot = 3, 1, -1 do
-                delAtma(player, atmaSlot)
-            end
+            player:messageSpecial(7448)
+            delAllAtma(player)
         else
+            local effect = player:getStatusEffect(xi.effect.ATMA, slot):getPower()
+            player:messageSpecial(7447, effect)
```
Comment (claywar) [inline scripts/globals/abyssea/atma.lua]: Use enum here, also, effect should most likely be renamed to effectPower to represent what it is

File: scripts/globals/abyssea/atma.lua [hunk 4]
```diff
@@ -432,29 +454,56 @@ end
...
             delAtma(player, slot)
         end
     elseif
         optionSelected == 3 and
         player:getCharVar("ABYSSEA_LAST_ATMA_INFUSED")
     then -- Reinfuse Atma
         local atmaReinfused = getAtmasFromMask(player:getCharVar("ABYSSEA_LAST_ATMA_INFUSED"))
+        local nbAtma = 0
```
Comment (claywar) [inline scripts/globals/abyssea/atma.lua]: Can you please use numAtma instead of nbAtma for readability?

File: scripts/globals/abyssea/atma.lua [hunk 5]
```diff
@@ -432,29 +454,56 @@ end
...
         for i = 1, 3 do
             if atmaReinfused[i] ~= 0 then
                 addAtma(player, atmaReinfused[i])
+                nbAtma = nbAtma + 1
             end
         end
+        player:messageSpecial(7454, atmaPrice * nbAtma)
```
Comment (claywar) [inline scripts/globals/abyssea/atma.lua]: enum

File: scripts/globals/abyssea/atma.lua [hunk 6]
```diff
@@ -432,29 +454,56 @@ end
...
+    elseif
+        optionSelected == 5
+    then -- infuse history
+        delAllAtma(player)
+        local mask = bit.rshift(option, 17)
+        local indexPos = { }
+        for i = 1, 12, 1 do
```
Comment (claywar) [inline scripts/globals/abyssea/atma.lua]: No need to specify default incrementer in this loop

**Comment by Chronos77** [comment]
Done !

**Review by Xaver-DaRed — APPROVED** [review]

**Pull request merged**